flake.nix

```diff
@@ -0,0 +1,45 @@
+{
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+    rust-overlay.url = "github:oxalica/rust-overlay";
+  };
+
+  outputs = { nixpkgs, rust-overlay, ... }:
+    let
+      system = "x86_64-linux";
+
+      pkgs = import nixpkgs {
+        inherit system;
+        overlays = [ rust-overlay.overlays.default ];
+      };
+
+      rust-toolchain = pkgs.rust-bin.fromRustupToolchainFile ./rust-toolchain.toml;
+    in
+    {
+      devShells.${system}.default = pkgs.mkShell rec {
+        packages = with pkgs; [
+          pkg-config
+          cmake
+          fontconfig
+
+          xorg.libX11
+          xorg.libXcursor
+          xorg.libXrandr
+          xorg.libXi
+          xorg.libxkbfile
+          xorg.xkbutils
+          xorg.xkbevd
+          xorg.libXScrnSaver
+          libxkbcommon
+
```

Comment:
Doesn't it need `wayland` as dependencies to compile for wayland? Please confirm.

Reply:
aye, good point! I looked it up in the [rio-package definition](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/terminal-emulators/rio/default.nix#L49).